File: code9.py
from math import log10
class Solution(object):
    def isPalindrome(self, x):

        dgt=log10(x)
        if x<0 : 
            return False
        else if x<10:
            return True
        mirror=0
        while x>mirror:
            mirror=mirror*10+x%10
            x=x//10
        print(x)
        print(mirror)
        return (dgt%2==0 and x==mirror) or (dgt%2==0 and mirror//10==x)
s=Solution()
print(s.isPalindrome(3))
print(s.isPalindrome(10))
print(s.isPalindrome(313))
print(s.isPalindrome(1221))



        
        

File: code9.cpp
/*
https://leetcode.com/problems/palindrome-number/
Given an integer x, return true if x is palindrome integer.

An integer is a palindrome when it reads the same backward as forward. For example, 121 is palindrome while 123 is not.
*/

#include <iostream>
#include <math.h>
using namespace std;

class Solution
{
public:
    bool isPalindrome(int x)
    {
        if (x < 0)
            return false;
        if (x == 0)
            return true;
        int val = floor(log10(x));
        // cout << val << endl;
        int divisor = pow(10, val);
        // cout << divisor << endl;
        while (x > 9)
        {
            int left = x / divisor;
            int right = x % 10;
            // cout << left << " " << right << endl;
            if (left != right)
                return false;
            x = x % divisor;
            x = x / 10;
            // cout << x << endl;
            divisor = divisor / 100;
            if(x==0) return true;
        }
        // cout<<divisor<<endl;
        return divisor==1;
    }
};

int main()
{
    int number;
    cin >> number;
    Solution s;
    bool res = s.isPalindrome(number);
    cout << res << endl;
}